code/game/machinery/cryo.dm

```diff
@@ -122,7 +122,14 @@
 	..()
 	if(autoeject)
 		if(occupant)
-			if(occupant.health >= 100)
+			var/mech_damage = 0
+			if(istype(occupant, /mob/living/carbon/human))
+				var/mob/living/carbon/human/H = occupant
+				for(var/obj/item/organ/external/O in H.organs)
+					if(O.status & ORGAN_ROBOT)
+						mech_damage += O.brute_dam
+						mech_damage += O.burn_dam
+			if(occupant.health >= (100 - mech_damage))

```

Comment:
This block seems rather applicable elsewhere - for example, for medbots checking if they can heal someone, and other automated healing routines in general. Perhaps this would be better-suited as a proc of its own, defined on both `/mob/living/` as `proc/get_mechanical_damage` which returns 0 on the base level, but on the `/mob/living/carbon/human` level would execute this code instead. You could even beat out the `istype` above with that, too!